### PR TITLE
Crud api

### PR DESCRIPTION
**Pull request by rdkerr**
Built out crud api, api endpoints, refactored paths, refactored postgres to maria db, created seeding script for mariadb that generates 10M listings and ~30M bookings.  It took about an hour for full seeding but it was handled by a VM with CPU/RAM restrictions.  Running on my homebox should be able to cut that time by at least half.  These restrictions limited the size of data I could hold in memory, I was able to get faster load times by using csvs.

Had an issue seeding when data contained multiple paragraphs so all descriptions limited to 1 at the moment.

### CODE REVIEW

File: db/mariadb/development/seed.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    IGNORE 1 LINES
+    (price, stars, reviews, cleaningFee, serviceFee, guests, minNights, title, address, highlights, introDesc, spaceDesc, guestDesc, otherDesc)
+    `)
+    await deleteCsv('listings');
+    await inc();
+  }
+
+  let end = Date.now();
+  let min = (start - end) * -1.666e-5;
+  let sec = Math.floor((min - Math.floor(min)) * 60);
```
Comment (ascherj) [inline db/mariadb/development/seed.js:96]: Interesting and clever way of calculating the runtime of seeding your scripts. What's happening on lines 95 and 96? How are you calculating the minutes and seconds?

File: server/index.js [hunk 2]
```diff
@@ -1,40 +1,63 @@
...
-  db.getBookings(req.params.listingId).then(records => {
-    console.log('server get bookings: ', records);
-    res.send(records);
+app.route('/rooms/bookings/:listingId')
+  .get((req, res) => {
+    db.getBookings(req.params.listingId)
+      .then(records => res.send(records))
+      .catch(err => console.log(err));
+  })
+  post((req, res) => {
```
Comment (ascherj) [inline server/index.js:41]: It looks like you missed a period before `post` on line 41.

**Review by ascherj — COMMENTED** [review]
Looks good so far Roger. Do you plan on adding any tests?

File: server/index.js (hunk 2 above)
Comment (rdkerr) [reply server/index.js:41]: Nice catch

File: db/mariadb/development/seed.js (hunk 1 above)
Comment (rdkerr) [reply db/mariadb/development/seed.js:96]: I forget where I found it,  start - end gives you the time difference in milliseconds.  That constant is enough to change the difference into minutes.  By subtracting the floor you get the leftover time and mulitply it by 60 to get the seconds.  I am going to add tests for the routes I have added with supertest

**Pull request merged**